### PR TITLE
gh-122686: pin attrs package (for hypothesis workflow)

### PR DESCRIPTION
**Pull request by skirpichev**
* Issue: gh-122686

### CODE REVIEW

**Comment by hugovk** [comment]
Do we still need to pin if upgrading to latest hypothesis==6.108.8?

If so, please can you add a comment as to why we're adding an attrs pin?

**Comment by skirpichev** [comment]
> Do we still need to pin if upgrading to latest hypothesis==6.108.8?

Yes, I think so.  See:
https://github.com/python/cpython/actions/runs/10246311678/job/28343157458

> If so, please can you add a comment as to why we're adding an attrs pin?

I did no debugging, but new release seems broken on the CPython main.  Pinning "just works", e.g. (see last commit):
https://github.com/python/cpython/pull/122682

I'll open issue for attrs and add a comment with reference.

File: Tools/requirements-hypothesis.txt [hunk 1]
```diff
@@ -1,4 +1,7 @@
 # Requirements file for hypothesis that
 # we use to run our property-based tests in CI.
 
+# see python-attrs/attrs#1326
```
Comment (vstinner) [inline Tools/requirements-hypothesis.txt]: Until the root cause is clearly identified, I would prefer to refer to the CPython issue instead:

```suggestion
# see https://github.com/python/cpython/issues/122686
```

**Review by vstinner — APPROVED** [review]
LGTM

**Comment by hugovk** [comment]
Thanks!

**Pull request merged**